### PR TITLE
CH-338: Initial check-in creating provisioner folder structure and co…

### PR DESCRIPTION
**Pull request by shaileshsvc**
…nfig files for install and config operations in node preparation and customer deployment phases.

Signed-off-by: shaileshsvc <shailesh.v.chaudhari@seagate.com>

# Problem Statement
https://jts.seagate.com/browse/CH-338 : Add workflow automation for customer deployment phase

# Design
https://seagate-systems.atlassian.net/wiki/spaces/CH/pages/1056079886/CORTX-Halo+Provisioner+HLD+WIP

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by mukhtar-inamdar — APPROVED** [review]

**Comment by ajaysrivas** [comment]
@shaileshsvc Please fix codacy issues.

File: provisioner/cmdconfig.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+import configparser
+import collections
+from const import FileType
+from error import CustomError
+
+class CmdConfig:
```
Comment (ajaysrivas) [inline provisioner/cmdconfig.py:25]: rename to ResourcesConfig

File: provisioner/config.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+#import yaml, csv
+import collections
+from const import FileType, ResourceType
+
+class Config:
```
Comment (ajaysrivas) [inline provisioner/config.py:24]: Rename to SiteConfig

File: provisioner/config/haloprov.cfg [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+[InfraMonitor]
```
Comment (ajaysrivas) [inline provisioner/config/haloprov.cfg:18]: change this to yaml format.

File: provisioner/config/nodeprep.cfg [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+[Ansible]
```
Comment (ajaysrivas) [inline provisioner/config/nodeprep.cfg:18]: Change to yaml config.

File: provisioner/error.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+class CustomError(Exception):
```
Comment (ajaysrivas) [inline provisioner/error.py:20]: This class may not be required.

File: provisioner/haloprov.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+#! /usr/bin/env python3
+
+from resource import Component
+from cmdconfig import CmdConfig
+from const import FileType
+
+def main():
+    cmdcfg = CmdConfig(FileType.INI,'./config/haloprov.cfg')
```
Comment (ajaysrivas) [inline provisioner/haloprov.py:27]: Use absolute path.

File: provisioner/cmdconfig.py (hunk 1 above)
Comment (shaileshsvc) [reply provisioner/cmdconfig.py]: Done

File: provisioner/config.py (hunk 2 above)
Comment (shaileshsvc) [reply provisioner/config.py]: Done

File: provisioner/config/haloprov.cfg (hunk 3 above)
Comment (shaileshsvc) [reply provisioner/config/haloprov.cfg]: Done

File: provisioner/config/nodeprep.cfg (hunk 4 above)
Comment (shaileshsvc) [reply provisioner/config/nodeprep.cfg]: Done

File: provisioner/error.py (hunk 5 above)
Comment (shaileshsvc) [reply provisioner/error.py:20]: Will remove in next iteration.

File: provisioner/haloprov.py (hunk 6 above)
Comment (shaileshsvc) [reply provisioner/haloprov.py]: Done

Comment (ajaysrivas) [reply provisioner/haloprov.py]: not required.

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**